modules/apps/foundation/frontend-theme/responsive-boat-theme/src/js/main.js

```diff
@@ -0,0 +1,39 @@
+
+// This function gets loaded when everything, including the portlets, is on the page
+
+Liferay.on('allPortletsReady', myPageJS);
+
+// Function Containing All of My Event Handlers
+
+function myPageJS() {
+	let nav = AUI.$('#navigation');
+	let navBtnDiv = AUI.$('#navigation div.nav-btn');
+	let portletParagraphs = AUI.$('#content .portlet-body p');
+
+	document.onscroll = function() {
+		// Toggle 'scrolled' class on nav based on scrolled location
+		if (window.scrollY >= 140 && !nav.hasClass('scrolled'))
+			nav.addClass('scrolled');
+		else if (window.scrollY < 140 && nav.hasClass('scrolled'))
+			nav.removeClass('scrolled');
```

Comment:
We should do a few things to optimize the JS.  Let me know what I did below to optimize it.

```
var hasCssScrolled = nav.hasClass('scrolled');
var winScrollY = windowScrollY;

var hasScrolled = hasCssScrolled && winSCrollY < 140;

nav.toggleClass('scrolled', !hasScrolled);
```

Reply:
Wow, much simpler. Looks like it storing the scroll position and class condition in variables and then test whether either is false.

It looked good to me, so I plugged this code into my theme. Then I saw the navbar flicker between states when I began to scroll down, but act normally when I scrolled past 140px. I made a truth table to try to learn more, and found what was causing the issue.
<table>
<thead>
<td><strong>Has class</strong></td>
<td><strong>ScrollY &lt; 140</strong></td>
<td><strong>Result</strong></td>
</thead>
<tbody>
<tr>
<td>false</td><td>true</td><td>true - add class</td>
<td>&lt;-- causes navbar to flicker</td>
</tr>
<tr><td>true</td><td>true</td><td>false - remove class</td></tr>
<tr><td>false</td><td>false</td><td>true - add class</td></tr>
<tr><td>true</td><td>false</td><td>true - add class</td></tr>
</tbody>
</table>

So I simplified the code a bit more to fix this:
var winScrollY = window.scrollY;
var hasScrolled = winScrollY < 140;

nav.toggleClass('scrolled', !hasScrolled);

Reply:
The other thing to consider with this scroll is that it will occur on every scroll.  We should look into deboucing the scroll so it does not occur on every scroll but rather after a certain amount of time is paused.

Reply:
I see what you mean. The browser is having to do a lot of calculations during each scroll burst, especially when there's a lot of content to keep track of. More to come in my next pull request.